sql/core/src/test/resources/sql-tests/inputs/postgreSQL/numeric.sql

```diff
@@ -884,6 +884,12 @@ SELECT '' AS to_char_10, to_char(val, 'S0999999999999999.999999999999999'), val
 -- SELECT '' AS to_char_35, to_char('100'::numeric, 'f"ool\"999');
 -- SELECT '' AS to_char_36, to_char('100'::numeric, 'f"ool\\"999');
 
+-- [SPARK-43815] Add to_varchar alias for to_char
+-- TO_VARCHAR()
+-- Alias of TO_CHAR()
+SELECT '' AS to_varchar_3, to_varchar(val, '9999999999999999.999999999999999PR'), val
```

Comment:
Could you move this test out from the `postgreSQL` folder, for instance to `sql-tests/inputs/charvarchar.sql`.

The tests like this were borrowed from PostgreSQL when we tried to achieve feature parity to the DBMS:
https://issues.apache.org/jira/browse/SPARK-27763
It is better to don't add new checks there.